src/core/ddsi/src/q_ddsi_discovery.c

```diff
@@ -1545,7 +1565,8 @@ int builtins_dqueue_handler (const struct nn_rsample_info *sampleinfo, const str
     if (pwr) guid = pwr->e.guid; else memset (&guid, 0, sizeof (guid));
     GVTRACE ("data(builtin, vendor %u.%u): "PGUIDFMT" #%"PRId64": ST%x %s/%s:%s%s\n",
              sampleinfo->rst->vendor.id[0], sampleinfo->rst->vendor.id[1],
-             PGUID (guid), sampleinfo->seq, statusinfo, d->topic->name, d->topic->type_name,
+             PGUID (guid), sampleinfo->seq, statusinfo,
+             pwr && (pwr->c.xqos->present & QP_TOPIC_NAME) ? pwr->c.xqos->topic_name : "", d->type->type_name,
```

Comment:
It'd be better to guarantee a topic name is present in all endpoints (I actually think it is already, or else it could easily be done because it were only ever the built-in ones that did not have a name).